File: js/task2/S2_ARR_04.js
function flatten1(arr) {
  return arr.reduce((result, item) => {
      return Array.isArray(item) 
          ? result.concat(item) 
          : result.concat([item]);
  }, []);
}

console.log(flatten1([1, [2,3], [4], 5]));   
console.log(flatten1([1, 2, 3]));            
console.log(flatten1([[1], [2], [3]]));        
console.log(flatten1([1, [2, [3]], 4]));       
console.log(flatten1([]));                     
console.log(flatten1([1, [2, 3], [4, [5, 6]]]));

File: python/task2/S2_ARR_04.py
def flatten1(arr):
  result = []
  for item in arr:
      if isinstance(item, list):
          result.extend(item)
      else:
          result.append(item)
  return result

print(flatten1([1, [2, 3], [4], 5]))
print(flatten1([1, 2, 3]))
print(flatten1([[1], [2], [3]]))
print(flatten1([1, [2, [3]], 4]))
print(flatten1([]))
print(flatten1([1, [2, 3], [4, [5, 6]]]))
